### PR TITLE
Misc Pipeline Workspace updates

### PR DESCRIPTION
**Pull request by andrewballantyne**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5747

**Analysis / Root cause & Solution Description**: 
Misc cleanup of missed items / skipped items / broken items through merges.

High level change list:
* Edit Pipeline Builder couldn't update existing workspaces to be optional if they were not already true
* Width of Params, Resources and Workspaces are equal on Pipeline Builder
* Params and Resources Details Tab have sticky footer (if applicable)
* Empty Directories are now visible on Details pages
* Cleaned up rendering of Pipeline Workspace Resources section to make it easier to consume (dev quality of life)
* Fixed Pipeline modals not showing optional workspaces as optional

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux 

Screen Shot 2021-04-09 at 11 34 21 AM
Screen Shot 2021-04-09 at 11 34 49 AM
Screen Shot 2021-04-09 at 11 35 39 AM
Screen Shot 2021-04-09 at 11 36 04 AM
Screen Shot 2021-04-09 at 11 40 40 AM

**Unit test coverage report**: 

Reworked tests:
```
 PASS  packages/pipelines-plugin/src/components/shared/workspaces/__tests__/WorkspaceResourcesSection.spec.tsx (20.598s)
  WorkspaceResourcesSection
    ✓ should handle nulls (4ms)
    ✓ should ignore emptyDirectories (1ms)
    ✓ should ignore unknown items
    ✓ should render the section if provided with one of the accepted types (8ms)
    ✓ should render known ResourceLinks (19ms)
```

New Tests:
```
 PASS  packages/pipelines-plugin/src/components/shared/workspaces/__tests__/EmptyDirectoriesSection.spec.tsx
  EmptyDirectoriesSection
    ✓ should handle nulls (6ms)
    ✓ should not render if there are no workspaces
    ✓ should not render if provided non-emptyDirectory workspaces (1ms)
    ✓ should render the empty directories section if there is at least one empty directory (4ms)
    ✓ should render a wrapper for each empty directory (2ms)
    ✓ should render just emptyDirectories when mixed in with others (1ms)

 PASS  packages/pipelines-plugin/src/components/shared/workspaces/__tests__/WorkspaceDefinitionList.spec.tsx
  WorkspaceDefinitionList
    ✓ should handle nulls (4ms)
    ✓ should not render anything when provided no workspaces
    ✓ should should render wrapper if at least one workspace is there (3ms)
    ✓ should have a wrapper for each workspace provided (2ms)
    ✓ should support optional workspaces (2ms)
    ✓ should support a mixture of both optional and not workspaces (1ms)
```

**Test setup:**

No real setup, just test workspaces in various scenarios:
1. Optional Workspaces (Start/Trigger modals + Pipeline Builder -- both new and edit)
2. Rendering of Workspaces on (Run) Details pages (Empty Directories + Workspace object resources)
3. Width of Pipelines / Resources / Workspaces should be consistent on the Pipeline Builder + Details tabs

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/cc @karthikjeeyar 
/cc @bgliwa01 
/cc @abhinandan13jan

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourcesSection.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  PipelineRunWorkspace,
+  VolumeTypeConfigMaps,
+  VolumeTypePVC,
+  VolumeTypeSecret,
+} from '../../../types';
+import { ResourceLink } from '@console/internal/components/utils';
+import { ConfigMapModel, PersistentVolumeClaimModel, SecretModel } from '@console/internal/models';
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourcesSection.tsx:10]: nit: sort imports

**Comment by karthikjeeyar** [comment]
Validations error message vertical spacing needs to be fixed.
image

**Comment by andrewballantyne** [comment]
> Verified all other points mentioned in the PR description, everything works as expected. I noticed the below issue while testing
> 
> Validations error message vertical spacing needs to be fixed.
> image

Yeah, that's unrelated to this work I'm pretty sure. I think that has to do with the max-height css I saw when I went to support multi-line components in MultiColumnField. If I can reproduce it outside of this PR, I'll log a bug.

File: frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourcesSection.tsx (hunk 1 above)
Comment (andrewballantyne) [reply frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourcesSection.tsx]: Sorted.

**Pull request merged**